theseus_gui/src/components/RowDisplay.vue

```diff
@@ -0,0 +1,215 @@
+<script setup>
+import { ChevronLeftIcon, ChevronRightIcon } from 'omorphia'
+import Instance from '@/components/ui/Instance.vue'
+import News from '@/components/ui/News.vue'
+import { onMounted, onUnmounted, ref } from 'vue'
+
+const props = defineProps({
+  instances: Array,
+  news: Array,
+  label: String,
+  canPaginate: Boolean,
+})
+
+const allowPagination = ref(false)
+const modsRow = ref(null)
+const newsRow = ref(null)
+const trendingRow = ref(null)
+
+const shouldRenderNormalInstances =
+  props.instances &&
+  props.instances?.length !== 0 &&
+  !props.canPaginate &&
+  props.instances?.some((i) => !i.trending)
+const shouldRenderNews = props.news && props.news?.length !== 0
+const shouldRenderTrending =
+  props.instances &&
+  props.instances?.length !== 0 &&
+  props.instances?.every((i) => i.trending || i.downloads > 50)
+
+const handlePaginationDisplay = () => {
+  let parentsRow
+  if (shouldRenderNormalInstances) parentsRow = modsRow.value
+  if (shouldRenderNews) parentsRow = newsRow.value
+  else if (shouldRenderTrending) parentsRow = trendingRow.value
+
+  if (!parentsRow) return
+
+  const children = parentsRow.children
+  const lastChild = children[children.length - 1]
+  const childBox = lastChild.getBoundingClientRect()
+
+  if (childBox.x + childBox.width > window.innerWidth) allowPagination.value = true
+  else allowPagination.value = false
+}
+
+onMounted(() => {
+  if (props.canPaginate) window.addEventListener('resize', handlePaginationDisplay)
+
+  // Check if pagination should be rendered on mount
+  handlePaginationDisplay()
+})
+
+onUnmounted(() => {
+  if (props.canPaginate) window.removeEventListener('resize', handlePaginationDisplay)
+})
+
+const handleLeftPage = () => {
+  if (shouldRenderNormalInstances) modsRow.value.scrollLeft -= 170
+  else if (shouldRenderNews) newsRow.value.scrollLeft -= 170
+  else if (shouldRenderTrending) trendingRow.value.scrollLeft -= 170
+}
+
+const handleRightPage = () => {
+  if (shouldRenderNormalInstances) modsRow.value.scrollLeft += 170
+  else if (shouldRenderNews) newsRow.value.scrollLeft += 170
+  else if (shouldRenderTrending) trendingRow.value.scrollLeft += 170
+}
+</script>
+
+<template>
+  <div class="row">
+    <div class="header">
+      <p>{{ props.label }}</p>
+      <hr aria-hidden="true" />
```

Comment:
The text is too dark (should use the color accent contrast) and the HR is too bright. It should be a moreof gray color